regression/goto-clang/Makefile

```diff
@@ -0,0 +1,20 @@
+default: tests.log
+
+test:
+	@../test.pl -p -c ../../../src/goto-cc/goto-cc
```

Comment:
Wouldn't this require 1) creating a symlink and 2) then invoking `goto-clang` (just as is done in the `goto-gcc` regression test folder)?

Reply:
No, should be fine -- goto-cc should detect that the back-end is clang, on the right platforms where clang is the default.

Reply:
So why do we need a separate directory then? It seems inconsistent with the `goto-cl` and `goto-gcc` directories, which specifically test the behaviour when the binary is named `goto-cl` or `goto-gcc`, respectively.

Reply:
We don't, fixed.